sway/input/keyboard.c

```diff
@@ -921,6 +919,11 @@ void sway_keyboard_destroy(struct sway_keyboard *keyboard) {
 	if (keyboard->seat_device->input_device->wlr_device->keyboard->group) {
 		sway_keyboard_group_remove(keyboard);
 	}
+	struct wlr_seat *wlr_seat = keyboard->seat_device->sway_seat->wlr_seat;
+	struct sway_input_device *device = keyboard->seat_device->input_device;
+	if (wlr_seat_get_keyboard(wlr_seat) == device->wlr_device->keyboard) {
+		wlr_seat_set_keyboard(wlr_seat, NULL);
```

Comment:
Hmm, this should happen automatically inside wlroots because the seat should listen to the keyboard's destroy event.

Reply:
The wlr_keyboard can still exist. This needs to occur when changing the keyboard changes seats in sway (which destroys and creates a new sway_keyboard) as well. The wlr_keyboard doesn't get destroyed in that case